QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,135 @@
+### Contributor Origin [2.ii]
+
+Must have confirmation of contributor origin (e.g. DCO, CLA, etc.).
+
+### Peer Review Policy [2.iii]
+
+All pull request have two peer reviews
+
+### Continuous Integration [2.iv]
+
+all pull request must pass CI on all [tier 1 platforms](https://www.ros.org/reps/rep-2000.html#support-tiers)
+
+### Documentation Policy [2.v]
+
+All pull request must resolve related documentation changes before merging
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+
+`class_loader` has a [feature list](http://wiki.ros.org/class_loader) and each item in the list links to the corresponding feature documentation.
+There is documentation for all of the features, and new features require documentation before being added.
+
+### Public API Documentation [3.ii]
+
+`class_loader` has embedded API documentation and it is generated using doxygen and is [hosted](http://wiki.ros.org/class_loader) along side the feature documentation.
+There is documentation for all of the public API, and new additions to the public API require documentation before being added.
+
+### License [3.iii]
+
+The license for `class_loader` is BSD-3-Clause, and a summary is in each source file, the type is declared in the `package.xml` manifest file, and a full copy of the license is in the [LICENSE](./LICENSE) file.
+
+There is an automated test which runs a linter (ament_copyright) that ensures each file has a license statement.
+
+### Copyright Statement [3.iv]
+
+The copyright holders each provide a statement of copyright in each source code file in `class_loader`.
+
+There is an automated test which runs a linter (ament_copyright) that ensures each file has at least one copyright statement.
+
+## Testing [4]
+
+### Feature Testing [4.i]
+
+Each feature in `class_loader` has corresponding tests which simulate typical usage, and they are located in the `test` directory.
+New features are required to have tests before being added.
+
+### Public API Testing [4.ii]
+
+Each part of the public API have tests, and new additions or changes to the public API require tests before being added.
+The tests aim to cover both typical usage and corner cases, but are quantified by contributing to code coverage.
+
+### Coverage [4.iii]
+
+`class_loader` follows the recommendations for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#coverage), and opts to use branch coverage instead of line coverage.
+
+This includes:
+
+- tracking and reporting branch coverage statistics
+- achieving and maintaining branch coverage at or above 95%
+- no lines are manually skipped in coverage calculations
+
+Changes are required to make a best effort to keep or increase coverage before being accepted, but decreases are allowed if properly justified and accepted by reviewers.
+
+Current coverage statistics can be viewed here:
+
+![](https://codecov.io/gh/ahcorde/class_loader/branch/ahcorde_code_coverage/graphs/sunburst.svg)
```

Comment:
This image isn't too helpful. Maybe you could run a ci_linux-coverage job and link that instead.